android/src/window.rs

```diff
@@ -0,0 +1,107 @@
+use jni::objects::JObject;
+use jni::JNIEnv;
+use maplibre::error::Error;
+use maplibre::io::scheduler::ScheduleMethod;
+use maplibre::io::source_client::HTTPClient;
+use ndk::native_window::NativeWindow;
+use raw_window_handle::{AndroidNdkHandle, RawWindowHandle};
+use std::marker::PhantomData;
+use std::thread::sleep;
+use std::time::Duration;
+
+use maplibre::map_state::MapState;
+use maplibre::window::{MapWindow, MapWindowConfig, Runnable, WindowSize};
+
+pub struct AndroidNativeWindow {
+    window: NativeWindow,
+}
+
+pub struct AndroidMapWindowConfig<'a> {
+    env: JNIEnv<'a>,
+    surface: JObject<'a>,
+}
+
+unsafe impl raw_window_handle::HasRawWindowHandle for AndroidNativeWindow {
+    fn raw_window_handle(&self) -> RawWindowHandle {
+        let mut handle = AndroidNdkHandle::empty();
+        handle.a_native_window = unsafe { self.window.ptr().as_mut() as *mut _ as *mut _ };
+        RawWindowHandle::AndroidNdk(handle)
+    }
+}
```

Comment:
Yes, let me add the bindings for this one directly on `ndk::NativeWindow` :tada:

Reply:
https://github.com/rust-windowing/android-ndk-rs/pull/274 Hopefully this simplifies things a bit for you, too :)